lib/dropdown/DropdownCtrl.js

```diff
@@ -0,0 +1,26 @@
+import angular from 'angular';
+
+var DropdownCtrlMod = angular.module('guDropdownCtrl', []);
+
+var DropdownCtrl = DropdownCtrlMod.controller('guDropdownCtrl', [
+  '$scope',
+  function DropdownCtrl(scope){
+
+      function getClassName() {
+        return scope.state === 'open' ? 'dropdown open' : 'dropdown closed';
```

Comment:
Actually, I find that having the classes in the template, at which point you can replace `state` with a simple boolean (isOpen?), makes it easier to distinguish state logic from the resulting CSS implementation. Keeping CSS classes in the template also helps group common concerns (HTML/CSS vs JS), rather than having to trawl back through deeper JS sources for the possible CSS values.